libcontainerd/container_windows.go

```diff
@@ -249,7 +249,13 @@ func (ctr *container) waitExit(process *process, isFirstProcessToStart bool) err
 						}
 						logrus.Error(err)
 					} else {
-						ctr.client.Create(ctr.containerID, ctr.ociSpec, ctr.options...)
+						if err := ctr.client.Create(ctr.containerID, ctr.ociSpec, ctr.options...); err != nil {
+							si.State = StateExit
```

Comment:
Is this a problem that we are setting `si.State` in a goroutine and also reading it in the parent below?
I see we are also setting it above as well.